src/app/code/community/Shopgate/Framework/Model/Payment/Abstract.php

```diff
@@ -304,11 +304,16 @@ public function createNewOrder($quote)
         if (!Mage::helper('shopgate/config')->getIsMagentoVersionLower15()) {
             $service->submitAll();
 
-            return $this->setOrder($service->getOrder());
+            $order = $this->setOrder($service->getOrder());
         } else {
             /** @noinspection PhpDeprecationInspection */
-            return $this->setOrder($service->submit());
+            $order = $this->setOrder($service->submit());
         }
+
+        Mage::dispatchEvent('checkout_submit_all_after', array('order' => $order, 'quote' => $quote));
+        Mage::dispatchEvent('sales_model_service_quote_submit_after', array('order' => $order, 'quote' => $quote));
```

Comment:
`submitAll` -> calls `submit` -> calls these events:
```
checkout_type_onepage_save_order
sales_model_service_quote_submit_before
sales_model_service_quote_submit_success
sales_model_service_quote_submit_failure
sales_model_service_quote_submit_after
```

So we don't want to be submitting it twice. This needs to be checked if that is the case in 1.4.* code, if not, just need to move it to the else clause. If it is -> needs to be removed.

`checkout_submit_all_after` seems to be correct

Reply:
it is correct that the call for 'sales_model_service_quote_submit_after' is unnecessary. I also checked Magento 1.4
double checked that the events are called in all our createNewOrder() functions (and only there)